docs/development/module/tests.md

```diff
@@ -6,104 +6,116 @@ Runtime tests allow you to verify the logic in your runtime module by mocking a
+	pub const TransferFee: u64 = 0;
+	pub const CreationFee: u64 = 0;
+}
 
-* `with_externalities`: A function that allows you to run logic on top of a `TestExternalities` storage.
+impl balances::Trait for TestRuntime {
+	type Balance = u64;
+	type OnNewAccount = ();
+	type OnFreeBalanceZero = ();
+	type Event = ();
+	type TransferPayment = ();
+	type DustRemoval = ();
+	type ExistentialDeposit = ExistentialDeposit;
+	type TransferFee = TransferFee;
+	type CreationFee = CreationFee;
+}
+```
 
-`TestExternalities` is generated from building storage off of a genesis configuration. That looks like:
+In the implementation of `system::Trait`, `AccountId` is set to `u64` just like `Balance` shown above. Place `(u64, u64)` pairs in the `balances` vec to seed `(AccountId, Balance)` pairs as the account balances.
 
 ```rust
-// This function basically just builds a genesis storage key/value store according to our desired mockup.
-fn new_test_ext() -> runtime_io::TestExternalities<Blake2Hasher> {
-	system::GenesisConfig::default().build_storage::<Test>().unwrap().into()
+pub fn build(self) -> runtime_io::TestExternalities {
+	GenesisConfig {
+		balances: Some(balances::GenesisConfig::<TestRuntime>{
+			balances: vec![
+				(1, 10),
+				(2, 20),
+				(3, 30),
+				(4, 40),
+				(5, 50),
+				(6, 60)
+			],
+			vesting: vec![],
+		}),
+	}.build_storage().unwrap().into()
 }
 ```
 
-#### Configuring Genesis
-
-You can set custom values for the genesis configuration for modules you include in your mock runtime.
-
-TODO
+Account 1 has balance 10, account 2 has balance 20, and so on.
 
 ### Block Production
 
-Your tests may require you to simulate block production to move forward the state of your chain.
-
-You will need to emulate the orchestration provided by the Executive module in order for your runtime to behave correctly.
+It will be useful to simulate block production to verify that expected behavior holds during block time dependent changes. 
 
-Usually this means executing the `on_initialize` and `on_finalize` functions of all the modules included in your test, and incrementing the block number tracked by the System module. This can be written as a function you include in your unit tests:
+A simple way of doing this increments the System module's block number between `on_initialize` and `on_finalize` calls from all modules with `System::block_number()` as the sole input.
 
 ```rust
 fn run_to_block(n: u64) {
 	while System::block_number() < n {
-		MyModule::on_finalize(System::block_number());
-		System::on_finalize(System::block_number());
-		System::set_block_number(System::block_number() + 1);
-		System::on_initialize(System::block_number());
-		MyModule::on_initialize(System::block_number());
+		let current_block = System::block_number();
+		ExampleModule::on_finalize(current_block);
+		System::on_finalize(current_block);
+		System::set_block_number(current_block + 1u64.into());
+		System::on_initialize(current_block);
```

Comment:
This is wrong. You call `on_initialize` on a old block number...

The syntax I provide does not run into this error.

Reply:
good catch, 
```suggestion
		System::on_initialize(current_block + 1.into());
```

Reply:
👎 You should use the old syntax and apply the change to both lines.

Reply:
To be clear, it is better to call `System::block_number()` four times in this method instead of calling it once and then using that value for each inner method call? 

Can you explain why the suggestion is wrong?

Reply:
Done

Reply:
It seems like minimizing storage reads is a good idea how about something like this:

```rust
let mut current_block = System::block_number();
while current_block < n {
  // on finalize calls
  current_block += 1;
  System::set_block_number(current_block);
  // on initialize calls
}
```